src/lib/list/selection-list.ts

```diff
@@ -0,0 +1,228 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  AfterContentInit,
+  Component,
+  ContentChild,
+  ContentChildren,
+  Directive,
+  ElementRef,
+  Input,
+  Optional,
+  QueryList,
+  Renderer2,
+  ViewEncapsulation,
+  ChangeDetectionStrategy,
+  OnDestroy,
+  EventEmitter,
+  Output,
+  ChangeDetectorRef
+} from '@angular/core';
+import {coerceBooleanProperty, MdLine, MdLineSetter, SelectionModel} from '../core';
+import {FocusKeyManager} from '../core/a11y/focus-key-manager';
+import {Subscription} from 'rxjs/Subscription';
+import {SPACE} from '../core/keyboard/keycodes';
+import {Focusable} from '../core/a11y/focus-key-manager';
+import {MdListOption} from './list-option';
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-selection-list, mat-selection-list',
+  host: {
+    'role': 'listbox',
+    '[attr.tabindex]': '_tabIndex',
+    'class': 'mat-selection-list',
+    '(focus)': 'focus()',
+    '(keydown)': 'keydown($event)'},
+  template: '<ng-content></ng-content>',
+  styleUrls: ['list.css'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class MdSelectionList implements AfterContentInit, OnDestroy {
+  private _disableRipple: boolean = false;
+
+  private _disabled: boolean = false;
+
+  /** Tab index for the selection-list. */
+  _tabIndex = 0;
+
+  /** Track which options we're listening to for focus/destruction. */
+  private _subscribed: WeakMap<MdListOption, boolean> = new WeakMap();
+
+  /** Subscription to tabbing out from the selection-list. */
+  private _tabOutSubscription: Subscription;
+
+  /** Subscription to option changes from the selection-list. */
+  private _optionSubscription: Subscription;
+
+  /** Whether or not the option is selectable. */
+  protected _selectable: boolean = true;
+
+  /** The FocusKeyManager which handles focus. */
+  _keyManager: FocusKeyManager;
+
+  /** The option components contained within this selection-list. */
+  @ContentChildren(MdListOption) options;
+
+  /** options which are selected. */
+  selectedOptions: SelectionModel<MdListOption> = new SelectionModel<MdListOption>(true);
+
+  /**
+   * Whether the ripple effect should be disabled on the list-items or not.
+   * This flag only has an effect for `mat-selection-list` components.
+   */
+  @Input()
+  get disableRipple() { return this._disableRipple; }
+  set disableRipple(value: boolean) { this._disableRipple = coerceBooleanProperty(value); }
+
+  /** Whether the selection-list is disabled */
+  @Input()
+  get disabled() { return this._disabled; }
+  set disabled(value: any) { this._disabled = coerceBooleanProperty(value); }
```

Comment:
Instead of defining the `disabled` getter/setter yourself, you can include it via mixin. Check out `button.ts` for an example of how to apply a mixin (there's some boilerplate involved but it ultimately reduces code duplication).